### PR TITLE
refactor: App Config Folder Structure

### PR DESCRIPTION
**Pull request by RohitRaj011**
# Description

refactor: appConfig folder structure
fix: eslint issues in appConfig components
refactor: AppConfig.tsx add context provider

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/Pages/Applications/DevtronApps/Details/AppConfigurations/AppComposeRouter.tsx [hunk 1]
```diff
@@ -123,17 +125,17 @@ export default function AppComposeRouter({
                     <Route
                         key={`${path}/${URLS.APP_CM_CONFIG}`}
                         path={`${path}/${URLS.APP_CM_CONFIG}/:name?`}
-                        render={(props) => <ConfigMapList isJobView={isJobView} isProtected={false} />}
+                        render={() => <ConfigMapList isJobView={isJobView} isProtected={false} />}
```
Comment (vivek-devtron) [inline src/Pages/Applications/DevtronApps/Details/AppConfigurations/AppComposeRouter.tsx:128]: We should change it to 
`<Route
                        key={`${path}/${URLS.APP_CM_CONFIG}`}
                        path={`${path}/${URLS.APP_CM_CONFIG}/:name?`} >
                        <ConfigMapList isJobView={isJobView} isProtected={false} />
                        </Route>`

File: src/Pages/Applications/DevtronApps/Details/AppConfigurations/AppConfig.utils.ts [hunk 2]
```diff
@@ -91,15 +92,15 @@ export const getCompletedStep = (
 }
 
 export const getNavItems = (
```
Comment (vivek-devtron) [inline src/Pages/Applications/DevtronApps/Details/AppConfigurations/AppConfig.utils.ts:94]: If possible we should move these long functions in utils

File: src/Pages/Applications/DevtronApps/Details/AppConfigurations/AppConfig.utils.ts [hunk 3]
```diff
@@ -248,10 +249,20 @@ export const getNavItems = (
...
+                isLocked: !_isUnlocked.envOverride,
             },
         ]
     }
 
     return { navItems }
 }
+
+export function isCIPipelineCreated(responseArr: AppConfigStatusResponseItem[]): boolean {
+    const ciPipeline = responseArr.find((item) => item.stageName === 'CI_PIPELINE')
```
Comment (vivek-devtron) [inline src/Pages/Applications/DevtronApps/Details/AppConfigurations/AppConfig.utils.ts:261]: Please add handling for null responseArr

File: src/Pages/Applications/DevtronApps/Details/AppConfigurations/AppConfig.utils.ts [hunk 4]
```diff
@@ -248,10 +249,20 @@ export const getNavItems = (
...
+    return ciPipeline.status
+}
+
+export function isCDPipelineCreated(responseArr: AppConfigStatusResponseItem[]): boolean {
+    const cdPipeline = responseArr.find((item) => item.stageName === 'CD_PIPELINE')
```
Comment (vivek-devtron) [inline src/Pages/Applications/DevtronApps/Details/AppConfigurations/AppConfig.utils.ts:266]: Same here

File: .eslintrc.js [hunk 5]
```diff
@@ -107,6 +107,8 @@ module.exports = {
             },
         ],
         'import/prefer-default-export': 'off',
+        'no-use-before-define': 'off',
+        '@typescript-eslint/no-use-before-define': 'error',
```
[the remark this answers is not in the document]
Comment (RohitRaj011) [reply .eslintrc.js:111]: discussed

File: src/Pages/Applications/DevtronApps/Details/AppConfigurations/AppConfig.utils.ts [hunk 6]
```diff
@@ -64,25 +65,25 @@ export const isUnlocked = (stage: string, isGitOpsConfigurationRequired?: boolea
 }
 
 export const getCompletedStep = (
-    isUnlocked: AppStageUnlockedType,
+    _isUnlocked: AppStageUnlockedType,
```
[the remark this answers is not in the document]
Comment (RohitRaj011) [reply src/Pages/Applications/DevtronApps/Details/AppConfigurations/AppConfig.utils.ts:68]: yes we already have isUnlocked declared above

File: src/Pages/Applications/DevtronApps/Details/AppConfigurations/AppConfig.utils.ts [hunk 7]
```diff
@@ -64,25 +65,25 @@ export const isUnlocked = (stage: string, isGitOpsConfigurationRequired?: boolea
...
     isJobView: boolean,
```
[the remark this answers is not in the document]
Comment (RohitRaj011) [reply src/Pages/Applications/DevtronApps/Details/AppConfigurations/AppConfig.utils.ts:69]: will be removed in further PRs

File: src/Pages/Applications/DevtronApps/Details/AppConfigurations/AppConfig.utils.ts [hunk 8]
```diff
@@ -64,25 +65,25 @@ export const isUnlocked = (stage: string, isGitOpsConfigurationRequired?: boolea
...
     isGitOpsConfigurationRequired: boolean,
 ): number => {
     if (isJobView) {
-        if (isUnlocked.workflowEditor) {
+        if (_isUnlocked.workflowEditor) {
             return 1
```
[the remark this answers is not in the document]
Comment (RohitRaj011) [reply src/Pages/Applications/DevtronApps/Details/AppConfigurations/AppConfig.utils.ts:74]: this is not a config from API, this is our internal handling

File: src/Pages/Applications/DevtronApps/Details/AppConfigurations/AppConfig.utils.ts (hunk 3 above)
Comment (RohitRaj011) [reply src/Pages/Applications/DevtronApps/Details/AppConfigurations/AppConfig.utils.ts:261]: function has been copied pasted from other location

Comment (RohitRaj011) [reply src/Pages/Applications/DevtronApps/Details/AppConfigurations/AppConfig.utils.ts:261]: discussed

File: src/Pages/Applications/DevtronApps/Details/AppConfigurations/AppConfig.utils.ts (hunk 4 above)
Comment (RohitRaj011) [reply src/Pages/Applications/DevtronApps/Details/AppConfigurations/AppConfig.utils.ts:266]: discussed

File: src/Pages/Applications/DevtronApps/Details/AppConfigurations/appConfig.type.ts [hunk 9]
```diff
@@ -44,25 +47,42 @@ export enum DEFAULT_LANDING_STAGE {
 
 export interface AppConfigProps {
     appName: string
+    resourceKind: Extract<ResourceKindType, ResourceKindType.devtronApplication | ResourceKindType.job>
     isJobView?: boolean
```
[the remark this answers is not in the document]
Comment (RohitRaj011) [reply src/Pages/Applications/DevtronApps/Details/AppConfigurations/appConfig.type.ts:51]: yes updated in further PR

File: src/Pages/GlobalConfigurations/Authorization/types.ts [hunk 10]
```diff
@@ -158,6 +158,10 @@ export interface UserRole {
      * If true, the user has super admin role
      */
     superAdmin: boolean
+    /**
+     * Role of the user
+     */
+    role: UserRoleType
```
[the remark this answers is not in the document]
Comment (RohitRaj011) [reply src/Pages/GlobalConfigurations/Authorization/types.ts:164]: discussed

File: src/Pages/Applications/DevtronApps/Details/AppConfigurations/AppConfig.tsx [hunk 11]
```diff
@@ -0,0 +1,508 @@
...
+            ) : (
+                <ConfirmationDialog>
+                    <ConfirmationDialog.Icon src={warn} />
+                    <ConfirmationDialog.Body title={`Cannot Delete ${isJobView ? 'job' : 'application'}`} />
+                    <p className="fs-13 cn-7 lh-1-54">
+                        Delete all pipelines and workflows before deleting this {isJobView ? 'job' : 'application'}.
+                    </p>
+                    <ConfirmationDialog.ButtonGroup>
+                        <button
+                            type="button"
```
[the remark this answers is not in the document]
Comment (RohitRaj011) [reply src/Pages/Applications/DevtronApps/Details/AppConfigurations/AppConfig.tsx:397]: discussed

File: src/Pages/Applications/DevtronApps/Details/AppConfigurations/AppConfig.tsx [hunk 12]
```diff
@@ -0,0 +1,508 @@
...
+    }
+
+    useEffect(() => {
+        // SET APP CONFIG DATA IN STATE
+        if (appConfigData) {
+            const [configStatusRes, workflowRes, { updatedEnvs, configProtections, isBaseConfigProtectionEnabled }] =
+                appConfigData
+            const { navItems, isCDPipeline, isCiPipeline, configs, redirectUrl, lastConfiguredStage } =
+                processConfigStatusData(configStatusRes.result)
+
```
[the remark this answers is not in the document]
Comment (RohitRaj011) [reply src/Pages/Applications/DevtronApps/Details/AppConfigurations/AppConfig.tsx:262]: will be done when in later PRs

**Review by eshankvaish — APPROVED** [review]

File: src/Pages/Applications/DevtronApps/Details/AppConfigurations/Navigation/EnvironmentOverrideRouter.tsx [hunk 13]
```diff
@@ -258,13 +262,30 @@ const EnvOverrideRoute = ({
     )
 }
 
-export default function EnvironmentOverrideRouter({
+const ValueContainer = (props: ValueContainerProps) => {
```
Comment (vivek-devtron) [inline src/Pages/Applications/DevtronApps/Details/AppConfigurations/Navigation/EnvironmentOverrideRouter.tsx:265]: We should extract this out

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**